Incorrect Types when using in a TypeScript ESM project

### 🕗 Version & Regression Information

1.2.0

### ⏯ Playground Link

<https://stackblitz.com/edit/node-3rxg4d>

1. Run `npm install`.
2. Run tsc compilation via `npm run build`. The error occurs.

### 💻 Code

```typescript
import invariant from 'tiny-invariant';

invariant(typeof 'foo' === 'string');
```

### 🙁 Actual behavior
I tried to run the code above from a **TypeScript ESM package**, this means:

- `"type": "module"` set in `package.json`
- `"module": "node16"` set in `tsconfig.json`
 
But in such a project, the code does not compile:

```sh
❯ npm run build
$ tsc
src/test.ts:3:1 - error TS2349: This expression is not callable.
  Type 'typeof import("/home/projects/node-3rxg4d/node_modules/tiny-invariant/dist/tiny-invariant")' has no call signatures.

invariant(typeof 'foo' === 'string');

Found 1 error in src/test.ts:3
```

Note that the compiled output does run as expected, so the only problem is the TypeScript compilation error.

### 🙂 Expected behavior

The code should compile.

--- a/package.json
+++ b/package.json
@@ -22,7 +22,7 @@
   "types": "dist/tiny-invariant.d.ts",
   "exports": {
     ".": {
-      "import": "./dist/node16/tiny-invariant.js",
+      "import": "./dist/esm/tiny-invariant.js",
       "default": "./dist/tiny-invariant.cjs.js"
     }
   },
@@ -60,10 +60,10 @@
     "build:clean": "rimraf dist",
     "build:flow": "cp src/tiny-invariant.js.flow dist/tiny-invariant.cjs.js.flow",
     "build:typescript": "tsc ./src/tiny-invariant.ts --emitDeclarationOnly --declaration --outDir ./dist",
-    "build:typescript:node16": "tsc ./src/tiny-invariant.ts --emitDeclarationOnly --declaration --outDir ./dist/node16",
-    "build:copy-esm-packagejson": "cp static/package-esm.json dist/node16/package.json",
+    "build:typescript:esm": "tsc ./src/tiny-invariant.ts --emitDeclarationOnly --declaration --outDir ./dist/esm",
+    "build:copy-esm-packagejson": "cp static/package-esm.json dist/esm/package.json",
     "build:dist": "yarn rollup --config rollup.config.js",
-    "build": "yarn build:clean && yarn build:dist && yarn build:typescript && yarn build:typescript:node16 && yarn build:copy-esm-packagejson",
+    "build": "yarn build:clean && yarn build:dist && yarn build:typescript && yarn build:typescript:esm && yarn build:copy-esm-packagejson",
     "prepublishOnly": "yarn build"
   },
   "devDependencies": {

--- a/package.json
+++ b/package.json
@@ -20,6 +20,12 @@
   "main": "dist/tiny-invariant.cjs.js",
   "module": "dist/tiny-invariant.esm.js",
   "types": "dist/tiny-invariant.d.ts",
+  "exports": {
+    ".": {
+      "import": "./dist/node16/tiny-invariant.js",
+      "default": "./dist/tiny-invariant.cjs.js"
+    }
+  },
   "sideEffects": false,
   "files": [
     "/dist",
@@ -54,8 +60,10 @@
     "build:clean": "rimraf dist",
     "build:flow": "cp src/tiny-invariant.js.flow dist/tiny-invariant.cjs.js.flow",
     "build:typescript": "tsc ./src/tiny-invariant.ts --emitDeclarationOnly --declaration --outDir ./dist",
+    "build:typescript:node16": "tsc ./src/tiny-invariant.ts --emitDeclarationOnly --declaration --outDir ./dist/node16",
+    "build:copy-esm-packagejson": "cp static/package-esm.json dist/node16/package.json",
     "build:dist": "yarn rollup --config rollup.config.js",
-    "build": "yarn build:clean && yarn build:dist && yarn build:typescript",
+    "build": "yarn build:clean && yarn build:dist && yarn build:typescript && yarn build:typescript:node16 && yarn build:copy-esm-packagejson",
     "prepublishOnly": "yarn build"
   },
   "devDependencies": {

--- a/rollup.config.js
+++ b/rollup.config.js
@@ -43,7 +43,7 @@ export default [
   {
     input,
     output: {
-      file: 'dist/node16/tiny-invariant.js',
+      file: 'dist/esm/tiny-invariant.js',
       format: 'esm',
     },
     plugins: [typescript({ module: 'ESNext' })],

--- a/rollup.config.js
+++ b/rollup.config.js
@@ -39,6 +39,15 @@ export default [
     },
     plugins: [typescript({ module: 'ESNext' })],
   },
+  // ESM build for "module": "node16" TypeScript projects (https://github.com/alexreardon/tiny-invariant/issues/144)
+  {
+    input,
+    output: {
+      file: 'dist/node16/tiny-invariant.js',
+      format: 'esm',
+    },
+    plugins: [typescript({ module: 'ESNext' })],
+  },
   // CommonJS build
   {
     input,

--- a/static/package-esm.json
+++ b/static/package-esm.json
@@ -1,0 +1,3 @@
+{
+  "type": "module"
+}

